From bf8e6b70276ddfd3e070964542020fe782c1eec8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Chigot?= <clement.chigot@atos.net>
Date: Thu, 27 Sep 2018 15:26:18 +0200
Subject: [PATCH] go/build, runtime/internal/sys: add GOOS=aix

This is the first commit of a series that will add AIX as an
operating system target for ppc64 architecture.

Updates #25893

Change-Id: I865b67a9c98277c11c1a56107be404ac5253277d
Reviewed-on: https://go-review.googlesource.com/138115
Run-TryBot: Brad Fitzpatrick <bradfitz@golang.org>
TryBot-Result: Gobot Gobot <gobot@golang.org>
Reviewed-by: Ian Lance Taylor <iant@golang.org>
---
 src/go/build/syslist.go                     |  2 +-
 src/runtime/internal/sys/zgoos_aix.go       | 21 +++++++++++++++++++++
 src/runtime/internal/sys/zgoos_android.go   |  1 +
 src/runtime/internal/sys/zgoos_darwin.go    |  1 +
 src/runtime/internal/sys/zgoos_dragonfly.go |  1 +
 src/runtime/internal/sys/zgoos_freebsd.go   |  1 +
 src/runtime/internal/sys/zgoos_linux.go     |  1 +
 src/runtime/internal/sys/zgoos_nacl.go      |  1 +
 src/runtime/internal/sys/zgoos_netbsd.go    |  1 +
 src/runtime/internal/sys/zgoos_openbsd.go   |  1 +
 src/runtime/internal/sys/zgoos_plan9.go     |  1 +
 src/runtime/internal/sys/zgoos_solaris.go   |  1 +
 src/runtime/internal/sys/zgoos_windows.go   |  1 +
 src/runtime/internal/sys/zgoos_zos.go       |  1 +
 14 files changed, 34 insertions(+), 1 deletion(-)
 create mode 100644 src/runtime/internal/sys/zgoos_aix.go

diff --git a/src/go/build/syslist.go b/src/go/build/syslist.go
index d7938fad54..597212b6d0 100644
--- a/src/go/build/syslist.go
+++ b/src/go/build/syslist.go
@@ -4,5 +4,5 @@
 
 package build
 
-const goosList = "android darwin dragonfly freebsd js linux nacl netbsd openbsd plan9 solaris windows zos "
+const goosList = "aix android darwin dragonfly freebsd js linux nacl netbsd openbsd plan9 solaris windows zos "
 const goarchList = "386 amd64 amd64p32 arm armbe arm64 arm64be ppc64 ppc64le mips mipsle mips64 mips64le mips64p32 mips64p32le ppc riscv riscv64 s390 s390x sparc sparc64 wasm "
diff --git a/src/runtime/internal/sys/zgoos_aix.go b/src/runtime/internal/sys/zgoos_aix.go
new file mode 100644
index 0000000000..9ce5b3434f
--- /dev/null
+++ b/src/runtime/internal/sys/zgoos_aix.go
@@ -0,0 +1,21 @@
+// Code generated by gengoos.go using 'go generate'. DO NOT EDIT.
+
+// +build aix
+
+package sys
+
+const GOOS = `aix`
+
+const GoosAndroid = 0
+const GoosAix = 1
+const GoosDarwin = 0
+const GoosDragonfly = 0
+const GoosFreebsd = 0
+const GoosLinux = 0
+const GoosNacl = 0
+const GoosNetbsd = 0
+const GoosOpenbsd = 0
+const GoosPlan9 = 0
+const GoosSolaris = 0
+const GoosWindows = 0
+const GoosZos = 0
diff --git a/src/runtime/internal/sys/zgoos_android.go b/src/runtime/internal/sys/zgoos_android.go
index bfdc37792e..36a5768ab6 100644
--- a/src/runtime/internal/sys/zgoos_android.go
+++ b/src/runtime/internal/sys/zgoos_android.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `android`
 
 const GoosAndroid = 1
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_darwin.go b/src/runtime/internal/sys/zgoos_darwin.go
index 1c4667f6de..10c0e88e9a 100644
--- a/src/runtime/internal/sys/zgoos_darwin.go
+++ b/src/runtime/internal/sys/zgoos_darwin.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `darwin`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 1
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_dragonfly.go b/src/runtime/internal/sys/zgoos_dragonfly.go
index 728bf6abe8..5cb47cb84e 100644
--- a/src/runtime/internal/sys/zgoos_dragonfly.go
+++ b/src/runtime/internal/sys/zgoos_dragonfly.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `dragonfly`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 1
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_freebsd.go b/src/runtime/internal/sys/zgoos_freebsd.go
index a8d659169b..470406ce5f 100644
--- a/src/runtime/internal/sys/zgoos_freebsd.go
+++ b/src/runtime/internal/sys/zgoos_freebsd.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `freebsd`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 1
diff --git a/src/runtime/internal/sys/zgoos_linux.go b/src/runtime/internal/sys/zgoos_linux.go
index 289400c612..76235b748c 100644
--- a/src/runtime/internal/sys/zgoos_linux.go
+++ b/src/runtime/internal/sys/zgoos_linux.go
@@ -8,6 +8,7 @@ package sys
 const GOOS = `linux`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_nacl.go b/src/runtime/internal/sys/zgoos_nacl.go
index 3fedb0a2c3..6d28b59667 100644
--- a/src/runtime/internal/sys/zgoos_nacl.go
+++ b/src/runtime/internal/sys/zgoos_nacl.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `nacl`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_netbsd.go b/src/runtime/internal/sys/zgoos_netbsd.go
index 3346e3711c..ef8d938ddb 100644
--- a/src/runtime/internal/sys/zgoos_netbsd.go
+++ b/src/runtime/internal/sys/zgoos_netbsd.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `netbsd`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_openbsd.go b/src/runtime/internal/sys/zgoos_openbsd.go
index 13c0323249..2e43847396 100644
--- a/src/runtime/internal/sys/zgoos_openbsd.go
+++ b/src/runtime/internal/sys/zgoos_openbsd.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `openbsd`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_plan9.go b/src/runtime/internal/sys/zgoos_plan9.go
index 6b2e977b5e..ed598dcaac 100644
--- a/src/runtime/internal/sys/zgoos_plan9.go
+++ b/src/runtime/internal/sys/zgoos_plan9.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `plan9`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_solaris.go b/src/runtime/internal/sys/zgoos_solaris.go
index cbf70f079a..fe690df6c2 100644
--- a/src/runtime/internal/sys/zgoos_solaris.go
+++ b/src/runtime/internal/sys/zgoos_solaris.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `solaris`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_windows.go b/src/runtime/internal/sys/zgoos_windows.go
index 70839ca793..ea7c43bdf4 100644
--- a/src/runtime/internal/sys/zgoos_windows.go
+++ b/src/runtime/internal/sys/zgoos_windows.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `windows`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
diff --git a/src/runtime/internal/sys/zgoos_zos.go b/src/runtime/internal/sys/zgoos_zos.go
index ecf449f703..d4027cf876 100644
--- a/src/runtime/internal/sys/zgoos_zos.go
+++ b/src/runtime/internal/sys/zgoos_zos.go
@@ -7,6 +7,7 @@ package sys
 const GOOS = `zos`
 
 const GoosAndroid = 0
+const GoosAix = 0
 const GoosDarwin = 0
 const GoosDragonfly = 0
 const GoosFreebsd = 0
-- 
2.17.1

